src/main/scala/uk/gov/nationalarchives/Lambda.scala

```diff
@@ -29,12 +29,22 @@ class Lambda extends RequestHandler[SQSEvent, Unit] {
         outputBucket = config.outputBucket
         fileProcessor = new FileProcessor(treInput.parameters.s3Bucket, outputBucket, batchRef, s3, randomUuidGenerator)
         fileNameToFileInfo <- fileProcessor.copyFilesFromDownloadToUploadBucket(treInput.parameters.s3Key)
-
         metadataFileInfo <- IO.fromOption(fileNameToFileInfo.get(s"$batchRef/TRE-$batchRef-metadata.json"))(
           new RuntimeException(s"Cannot find metadata for $batchRef")
         )
+
         treMetadata <- fileProcessor.readJsonFromPackage(metadataFileInfo.id)
-        parsedUri <- fileProcessor.parseUri(treMetadata.parameters.PARSER.uri)
+        uri = treMetadata.parameters.PARSER.uri
+        fileName = treMetadata.parameters.PARSER.name
+        _ <- {
+          val uriContainsPressSummary = uri.getOrElse("").contains("/press-summary")
```

Comment:
This is a style thing so feel free to ignore but what do you think about having:
```scala
          val uriContainsPressSummary = uri.exists(_.contains("/press-summary"))
          val fileNameDoesNotStartWithPressSummaryOf = !fileName.exists(_.startsWith("Press Summary of "))
```
It gets rid of the getOrElse("") at any rate. 

Also, what do you think about moving this block to a method in `FileProcessor` Most of the other lines in this for comprehension are single method calls and it can make it easier to read if you don't put too much logic in here.

Reply:
1. Ah, yes; that solution looks a lot cleaner 👍 
2. Regarding the 2nd solution, I wanted to have a call with you because I was thinking of lightly refactoring this `FileProcessor` file a little, as it's doing quite a bit which means that the `handleRequest` is hiding a lot of information making it harder to follow/debug. Regarding what you suggested, what if we created a `UriProcessor` Class, then I added a method with this press summary code in it and also moved the `parseUri` method from the `FileProcessor` Class to the `UriProcessor` Class (since parseUri is not doing any file processing per se)?